cognite/neat/_rules/models/dms/_validation.py

```diff
@@ -54,13 +55,19 @@ class DMSValidation:
     # For example, changing the filter is allowed, but changing the properties is not.
     changeable_view_attributes: ClassVar[set[str]] = {"filter"}
 
-    def __init__(self, rules: DMSRules, client: NeatClient | None = None) -> None:
+    def __init__(
+        self,
+        rules: DMSRules,
+        client: NeatClient | None = None,
+        read_info_by_spreadsheet: dict[str, SpreadsheetRead] | None = None,
+    ) -> None:
         self._rules = rules
         self._client = client
         self._metadata = rules.metadata
         self._properties = rules.properties
         self._containers = rules.containers
         self._views = rules.views
+        self._read_info_by_spreadsheet = read_info_by_spreadsheet or {}
```

Comment:
Adjust row numbers directly in the Validtion, instead of using the context manager. Making it more explicit.